app/assets/javascripts/discourse/tests/integration/components/user-menu/bookmarks-list-test.js

```diff
@@ -0,0 +1,78 @@
+import { module, test } from "qunit";
+import { setupRenderingTest } from "discourse/tests/helpers/component-test";
+import { exists, query, queryAll } from "discourse/tests/helpers/qunit-helpers";
+import { render, settled } from "@ember/test-helpers";
+import { NOTIFICATION_TYPES } from "discourse/tests/fixtures/concerns/notification-types";
+import { hbs } from "ember-cli-htmlbars";
+import pretender, { response } from "discourse/tests/helpers/create-pretender";
+import I18n from "I18n";
+
+module(
+  "Integration | Component | user-menu | bookmarks-list",
+  function (hooks) {
+    setupRenderingTest(hooks);
+
+    const template = hbs`<UserMenu::BookmarksList/>`;
+
+    test("renders notifications on top and bookmarks on bottom", async function (assert) {
+      await render(template);
+      const items = queryAll("ul li");
+
+      assert.strictEqual(items.length, 2);
+
+      assert.ok(items[0].classList.contains("notification"));
+      assert.ok(items[0].classList.contains("unread"));
+      assert.ok(items[0].classList.contains("bookmark-reminder"));
+
+      assert.ok(items[1].classList.contains("bookmark"));
+    });
+
+    test("show all link", async function (assert) {
+      await render(template);
+      const link = query(".panel-body-bottom .show-all");
+      assert.ok(
+        link.href.endsWith("/u/eviltrout/activity/bookmarks"),
+        "links to the bookmarks page"
+      );
+      assert.strictEqual(
+        link.title,
+        I18n.t("user_menu.view_all_bookmarks"),
+        "has a title"
+      );
+    });
+
+    test("dismiss button", async function (assert) {
```

Comment:
I think it'll also be helpful to test what bookmarks are shown after the dismiss button is used. Do we remove the reminders from the list? Or do we trigger another request to fetch all the bookmarks again?

Reply:
Note this behavior is already tested in `acceptance/user-menu-test.js` (we re-fetch the bookmarks after clicking dismiss).

Reply:
Ah icic.